### PR TITLE
Add lock for escalation handling subtask

### PR DESCRIPTION
**Pull request by lamafab**
Fixes https://github.com/w3f/SecOps/issues/117

Haven't tested it, but should work :)

### CODE REVIEW

**Review by Nexus2k — APPROVED** [review]

**Comment by ironoa** [comment]
chart version upgrade is missing

File: src/processor.rs [hunk 1]
```diff
@@ -159,17 +164,30 @@ impl Actor for Processor {
...
-                            Err(err) => error!("{:?}", err),
-                        }
-                    });
+                    let is_locked = lock.load(Ordering::Relaxed);
+
+                    if !is_locked {
+                        // Acquire new handlers for async task.
+                        let db = Arc::clone(&db);
+                        let lock = Arc::clone(&lock);
+
```
Comment (ironoa) [inline src/processor.rs]: just as a double check: can a thread take over here and pass the lock even if it is not supposed to ?

File: src/processor.rs [hunk 2]
```diff
@@ -159,15 +162,25 @@ impl Actor for Processor {
...
                 move |_proc, _ctx| {
+                    // Acquire new handles for async task.
                     let db = Arc::clone(&db);
+                    let lock = Arc::clone(&lock);
 
                     actix::spawn(async move {
-                        match local(db, escalation_window).await {
-                            Ok(_) => {}
-                            Err(err) => error!("{:?}", err),
+                        // Immediately exists if the lock cannot be acquired.
```
Comment (Nexus2k) [inline src/processor.rs]: exits?

Comment (lamafab) [reply src/processor.rs]: Yes, fixed :)

**Review by ironoa — APPROVED** [review]

**Pull request merged**